test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.ts

```diff
@@ -161,10 +162,11 @@ describe('Confirmation Redesign Contract Interaction Component', function () {
           await confirmContractDeploymentTransaction(driver);
 
           await createDepositTransaction(driver);
-          await driver.waitUntilXWindowHandles(3);
-          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
```

Comment:
This spec has several points that need to be changed, as a first step of fixing the tests:
- we should leverage fixtures for having the nonce enabled, instead of going to settings
```
.withPreferencesController({
      preferences: { redesignedConfirmationsEnabled: true },
      useNonceField: true, // enable nonce here
})
```
- we should remove the step of going to settings and enabling nonce
- we should remove the step of deploying a contract (it's already deployed)
- we should remove the step of confirming the contract deployment (it's already deployed)


```
// all 3 steps should be removed
await toggleOnCustomNonce(driver);
await createContractDeploymentTransaction(driver);
await confirmContractDeploymentTransaction(driver);
```

- we should login with balance validation to mitigate the flakiness point where the dialog does not open, as the wallet is not ready  `await logInWithBalanceValidation(driver);`
- we can add the ` await driver.waitUntilXWindowHandles(3);` before switching to the dialog window

Once this is done, we see that the test now fails consistently on the edit nonce step. This is actually a legit failure as I don't see the nonce field in the transaction screen, when we have the redesign toggle enabled, so it's something to fix on the app level.

https://github.com/user-attachments/assets/43aef96c-17dd-4233-83f3-3299c1394e32




Here is the e2e with the changes implemented and the current legit failure:

https://github.com/user-attachments/assets/33915b1d-848e-44e2-95d6-b78665a60f03